### ISSUE TITLE
New search filter "no_external_id" and "any_external_id"

### ISSUE THREAD

**Issue by blacksd** [firefly-iii/firefly-iii#6605]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)
- [X] My request is not listed as [a very good idea, but unfortunately...](https://docs.firefly-iii.org/firefly-iii/about-firefly-iii/what-its-not/)
- [X] I've used [the search](https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue) and this has not been discussed before.

### Description

I'd love to have a couple of  `_external_id` related boolean flags, as the current `no_external_url:true` and `any_external_url:true`.
This would ease up a lot of the maintenance for folks that - like me - are planning on using an external gateway to keep data imported but also have a lot of older data backlog that must be imported separately from CSV files. 
The former have an `external_id`, the latter don't; since some duplicates will happen, it would be very nice to have some filtering aids to identify and purge the duplicates.

### Solution

Add to the search some general catch-all conditions (well, just one and its negation) and make it available as search filter

### What are alternatives?

I tried messing with

```
external_id_is:query
external_id_contains:query
external_id_starts:query
external_id_ends:query
``` 

but nothing fits my case

**Comment by JC5** [comment]
Sure thing, it's on the list!

**Comment by JC5** [comment]
Fixed in 5.8.0. Alpha version will be released soon.
